activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -44,17 +44,17 @@ def sql(event)
     private
 
       def type_casted_binds(binds, casted_binds)
-        casted_binds || binds.map { |attr| type_cast attr.value_for_database }
+        casted_binds || ActiveRecord::Base.connection.send(:type_casted_binds, binds)
```

Comment:
Can we make this method public but private API? We prefer to avoid using `send` inside the framework.